client/state/analytics/middleware.js

```diff
@@ -32,45 +32,45 @@ const eventServices = {
 	adwords: ( { properties } ) => trackCustomAdWordsRemarketingEvent( properties ),
 };
 
-// Whitelists specific parameters to avoid polluting page view events
-const PAGE_VIEW_SERVICES_ALLOWED_PARAMS = [ 'client_id' ];
-
 const pageViewServices = {
 	ga: ( { url, title } ) => analytics.ga.recordPageView( url, title ),
-	default: ( { url, title, ...params } ) =>
-		analytics.pageView.record( url, title, pick( params, PAGE_VIEW_SERVICES_ALLOWED_PARAMS ) ),
+	default: ( { url, title, ...params } ) => analytics.pageView.record( url, title, params ),
 };
 
-const loadTrackingTool = ( trackingTool, state ) => {
+const loadTrackingTool = ( { trackingTool }, state ) => {
 	if ( trackingTool === 'HotJar' && ! isTracking( state, 'HotJar' ) ) {
 		analytics.hotjar.addHotJarScript();
 	}
 };
 
 const statBump = ( { group, name } ) => analytics.mc.bumpStat( group, name );
 
+const setAnonymousUserId = ( { anonId } ) => analytics.tracks.setAnonymousUserId( anonId );
+
+const setOptOut = ( { isOptingOut } ) => analytics.tracks.setOptOut( isOptingOut );
+
 export const dispatcher = ( { meta: { analytics: analyticsMeta } }, state ) => {
 	analyticsMeta.forEach( ( { type, payload } ) => {
-		const { service = 'default' } = payload;
+		const { service = 'default', ...params } = payload;
 
 		switch ( type ) {
 			case ANALYTICS_EVENT_RECORD:
-				return invoke( eventServices, service, payload );
+				return invoke( eventServices, service, params );
 
 			case ANALYTICS_PAGE_VIEW_RECORD:
-				return invoke( pageViewServices, service, payload );
+				return invoke( pageViewServices, service, params );
 
 			case ANALYTICS_STAT_BUMP:
-				return statBump( payload );
+				return statBump( params );
 
 			case ANALYTICS_TRACKING_ON:
-				return loadTrackingTool( payload, state );
+				return loadTrackingTool( params, state );
 
 			case ANALYTICS_TRACKS_ANONID_SET:
-				return analytics.tracks.setAnonymousUserId( payload );
+				return setAnonymousUserId( params );
```

Comment:
Careful! The original payload param isn't an object.
```
export const setTracksAnonymousUserId = anonId => ( {
	type: ANALYTICS_TRACKS_ANONID_SET,
	meta: {
		analytics: [
			{
				type: ANALYTICS_TRACKS_ANONID_SET,
				payload: anonId,
			},
		],
	},
} );
```

Reply:
It's not clean, but we'll want to pass through the original payload and keep the left side for these two.

For example:
```jsx
const { b = 'default' , ...rest } = 5678;
// b = default
// rest = {};
// we lose the original paylod
```